src/ucp/wireup/select.c

```diff
@@ -1577,16 +1577,18 @@ ucp_wireup_construct_lanes(const ucp_wireup_select_params_t *select_params,
             key->tag_lane = lane;
         }
 
-        /* add lanes to ep_check map */
-        dev_index = context->tl_rscs[key->lanes[lane].rsc_index].dev_index;
-        ucs_assert(dev_index < (sizeof(dev_map_used) * 8));
-        iface_attr = ucp_worker_iface_get_attr(worker,
-                                               key->lanes[lane].rsc_index);
-        if (!(UCS_BIT(dev_index) & dev_map_used) &&
-             (iface_attr->cap.flags & UCT_IFACE_FLAG_EP_CHECK)) {
-            ucs_assert(!(key->ep_check_map & UCS_BIT(lane)));
-            key->ep_check_map |= UCS_BIT(lane);
-            dev_map_used      |= UCS_BIT(dev_index);
+        if (key->err_mode != UCP_ERR_HANDLING_MODE_NONE) {
+            /* add lanes to ep_check map */
+            dev_index = context->tl_rscs[key->lanes[lane].rsc_index].dev_index;
+            ucs_assert(dev_index < (sizeof(dev_map_used) * 8));
+            iface_attr = ucp_worker_iface_get_attr(worker,
+                                                   key->lanes[lane].rsc_index);
+            if (!(UCS_BIT(dev_index) & dev_map_used) &&
+                 (iface_attr->cap.flags & UCT_IFACE_FLAG_EP_CHECK)) {
+                ucs_assert(!(key->ep_check_map & UCS_BIT(lane)));
+                key->ep_check_map |= UCS_BIT(lane);
```

Comment:
does it make sense to have some lanes with keep_alive and others without?

Reply:
yes, here is created map for lanes with ep_check capability

Reply:
ok what if am lane supports ep_check but RNDV dont? I mean that all lanes (or some specific types?) must to support `ep_check` to guarantee workability

Reply:
we need just one keepalive lane per device

Reply:
added todo